crates/core/component/dex/src/component/position_manager.rs

```diff
@@ -81,11 +81,11 @@ pub trait PositionRead: StateRead {
         self.get(&state_key::position_by_id(id)).await
     }
 
-    async fn check_position_by_id(&self, id: &position::Id) -> Result<bool> {
-        Ok(self
-            .get_raw(&state_key::position_by_id(id))
-            .await?
-            .is_some())
+    async fn check_position_by_id(&self, id: &position::Id) -> bool {
```

Comment:
Why the change to a raw bool return type? With this change, we must use `expect` which will panic rather than return an error which can be handled more gracefully by callers.

Reply:
This is infallible unless the application state is corrupted, in that case there are no way to recover, so we do want to stop everything. So in effect, the `Result` here is superfluous, on the other hand it has a real cost of making the API unwieldy.